specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json

```diff
@@ -0,0 +1,1222 @@
+          "enum": [
+            "Accepted",
+            "Creating",
+            "Updating",
+            "Failed",
+            "Succeeded",
+            "Deleting",
+            "Canceled",
+            "Migrating"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          },
+          "description": "State of provisioning of the AzureBareMetalStorageInstance"
+        },
+        "offeringType": {
+          "type": "string",
+          "description": "the offering type for which the resource is getting provisioned"
+        },
+        "storageType": {
+          "type": "string",
+          "description": "the storage protocol for which the resource is getting provisioned"
+        },
+        "generation": {
+          "type": "string",
+          "description": "the kind of storage instance"
+        },
+        "hardwareType": {
+          "type": "string",
+          "description": "the hardware type of the storage instance"
+        },
+        "workloadType": {
+          "type": "string",
+          "description": "the workload for which the resource is getting provisioned"
+        },
+        "storageBillingProperties": {
+          "$ref": "#/definitions/StorageBillingProperties",
+          "description": "the billing related information for the resource"
+        }
+      },
+      "description": "described the storage properties of the azure bare metal storage instance"
+    },
+    "StorageBillingProperties": {
+      "type": "object",
+      "properties": {
+        "billingMode": {
+          "type": "string",
+          "description": "the billing mode for the storage instance"
+        },
+        "azureBareMetalStorageInstanceSize": {
+          "type": "string",
+          "description": "the SKU type that is provisioned"
+        }
+      },
+      "description": "Describes the billing related details of the AzureBareMetalStorageInstance."
+    },
+    "Tags": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Tags field of the AzureBareMetal/AzureBareMetaStorage instance."
+        }
+      },
+      "description": "Tags field of the AzureBareMetal/AzureBareMetaStorage instance."
+    },
+    "ForceFlag": {
+      "type": "object",
+      "properties": {
+        "force": {
+          "type": "boolean",
+          "description": "Whether to force restart by shutting all processes."
+        }
+      },
+      "description": "The flag empowers the server with the ability to forcefully terminate and halt any existing processes that may be running on the server"
+    },
+    "OperationStatus": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique Operation Status Identifier."
+        },
+        "status": {
```

Comment:
Consider model it as an enums if the property has a finite set of values?
refs: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi015-avoid-modelling-a-property-as-string-type-when-it-should-be-an-enums